stag-library/src/main/java/com/vimeo/stag/UseStag.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License (MIT)
+ * <p/>
+ * Copyright (c) 2016 Vimeo
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.stag;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation to tell Stag that all the member variables of the class should be processed.
+ * <p/>
+ * If the class is annotated, Stag will generate a TypeAdapter for that class.
+ * Stag does not generate TypeAdapters for abstract classes.
+ * <p/>
+ */
+@Target({ElementType.TYPE})
+public @interface UseStag {
```

Comment:
I am a little confused on why there is the need to add a ```@UseStag``` annotation. The current ```@GsonAdapterKey``` annotation can be applied to the class level and to member variables. In the PR description, it mentions that ```UseStag``` can help in the ease of integration so that consumers need only annotate a model, but that is already in place.

Perhaps ```@UseStag``` can replace the class level ```@GsonAdapterKey``` while the latter is kept in place on a variable level? Although they would perform the same thing, it may be more clear to use.

Having said that, if a new annotation does not bring additional and/or different benefits than the current annotation, then I would be slightly opposed to a name change just for the sake of a name change. From a consumption standpoint I hate when library releases make you refactor all of your code. I would prefer to keep my code in tact but reap in the benefits from library updates - even major updates. Considering this library is still in its infancy yet has many adopters, I would personally prefer that we stick with the current annotation name, if possible.